pkg/config/draftconfig.go

```diff
@@ -1,15 +1,16 @@
 package config
 
 import (
+	"errors"
+
 	log "github.com/sirupsen/logrus"
 )
 
 // TODO: remove Name Overrides since we don't need them anymore
 type DraftConfig struct {
-	DisplayName      string              `yaml:"displayName"`
-	NameOverrides    []FileNameOverride  `yaml:"nameOverrides"`
-	Variables        []BuilderVar        `yaml:"variables"`
-	VariableDefaults []BuilderVarDefault `yaml:"variableDefaults"`
+	DisplayName   string                `yaml:"displayName"`
+	NameOverrides []FileNameOverride    `yaml:"nameOverrides"`
+	Variables     map[string]BuilderVar `yaml:"variables"`
```

Comment:
not sure I agree with making the name the key. Generally, you want to group things together so the name would be another field under `BuilderVar` and this would remain a slice.

Reply:
So, the main reason why I introduced a map here was for the simplicity of the DevHub integration. Currently on DevHub, a variable is declared for each expected envArg's value which is fetched from the devHubTypes.Workflow struct built from the protos. Initially, I was able to create a hashmap for the slice version of BuilderVars where the key was the name and the value was the index. Therefore, once I had the variable and was ready to put it into it's respective BuilderVar, I could index the slice in O(1) time rather than iterating through it until I found the right name. This redundancy of using a hashmap is what led to the decision to change the overall data structure from a slice to a map. Let me know if you still consider the slice to be the better implementation though!

Reply:
What method is DevHub using to interface with this? 

Just because the struct for this doesn't use a map doesn't mean that we can convert it into a map at the entrypoint for extending Draft

Reply:
We're updating the interface of the templating of draft to take a DraftConfig for template definitions rather than a map[string][string] vals.  Having a map[VariableName]BuilderVar within the draft config makes some things a little easier, like indexing specific variables for user input. This can also be applied in prompt/flag input as well.

Reply:
DevHub isn't currently using a method that utilizes Draft, it only exists on my local machine. I used the GetStarterWorkflow() on the starterworkflows.go file as a base for how to recreate it using the DraftConfig struct. Sorry if my explanation was confusing. I'm not converting the slice of BuilderVars into a map, I'm looping through the slice, taking each variable's name and it's index in the slice, and then plugging that information into a separate map that I instantiated within the function.

Reply:
Changed it back to a slice as we can use a resource field for validation/prompt selection, and this allows the prompts to always come out in a specified order.